47.md

```diff
@@ -105,18 +86,60 @@ Response:
 ```jsonc
 {
     "result_type": "pay_invoice",
-    "result": { 
+    "result": {
         "preimage": "0123456789abcdef..." // preimage of the payment
     }
 }
 ```
 
+## Authorization
+
+Each **client** uses a randomly generated secret to sign the events. The **wallet service** uses the corresponding public key to verify the event and authorize the action.
+
+- Users can revoke the authorization at any time on the **wallet service** or can define arbitrary constraints (eg. budgets).
+- The key is harder to leak since it is not shown to the user and backed up.
+- It improves privacy because the user's main key would not be linked to their payments
+
+
 ## Example pay invoice flow
 
 0. The user scans the QR code generated by the **wallet service** with their **client** application, they follow a `nostr+walletconnect:` deeplink or configure the connection details manually.
-1. **client** sends an event to with **wallet service** service with kind `23194`. The content is a `pay_invoice` request. The private key is the secret from the connection string above.
+1. **client** sends an event to with **wallet service** service with kind `23194`. The content is a `pay_invoice` request. The private key is the secret of the connection to which the **wallet service** has the corresponding public key.
 2. **wallet service** verifies that the author's key is authorized to perform the payment, decrypts the payload and sends the payment.
 3. **wallet service** responds to the event by sending an event with kind `23195` and content being a response either containing an error message or a preimage.
 
 ## Using a dedicated relay
 This NIP does not specify any requirements on the type of relays used. However, if the user is using a custodial service it might make sense to use a relay that is hosted by the custodial service. The relay may then enforce authentication to prevent metadata leaks. Not depending on a 3rd party relay would also improve reliability in this case.
+
+## Connection URL
+For maximum interoperability the connection string is standardized as URL scheme to share required connection information between the **client** and the **wallet service**
+
+The scheme is nostr+walletconnect:
+
+The base path it's hex-encoded **wallet service** `pubkey` with the following query string parameters:
+
+- `relay` URL of the relay where the **wallet service** is connected and will be listening for events. May be more than one.
+- `secret` 32-byte randomly generated hex encoded string. The **client** MUST use this to sign events and encrypt payloads when communicating with the **wallet service**.
+- `pubkey` The hex-encoded pubkey for the `secret`
```

Comment:
isn't this the pubkey of the wallet service?

Reply:
I guess this is these are depending on the flow.